#VRML_SIM R2021a utf8
WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation 0.16338437931567823 0.9534616092586478 0.2534097556637082 4.243965065915699
  position -3.453905344238553 2.0343632641264504 -1.380209658126575
}
TexturedBackground {
}
TexturedBackgroundLight {
}
DEF FLOOR RectangleArena {
  floorSize 2 2
  wallHeight 1e-07
}
Robot {
  translation -0.611831 0.06 -0.496105
  rotation 0 1 0 1.5707
  children [
    DEF BODY_GROUP Group {
      children [
        DEF BODY Shape {
          appearance PBRAppearance {
            baseColor 0.917647 0.145098 0.145098
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.1 0.05 0.2
          }
        }
        Transform {
          translation 0 0 0.12
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.917647 0.145098 0.145098
                metalness 0
              }
              geometry Box {
                size 0.02 0.02 0.04
              }
            }
          ]
        }
      ]
    }
    DEF wheelLeftF HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.06 0 0.05
      }
      device [
        PositionSensor {
          name "ps"
        }
        RotationalMotor {
          name "leftF"
        }
      ]
      endPoint Solid {
        translation 0.06 0 0.05
        rotation 0 1.5700895750559044e-16 1 1.5707999999999989
        children [
          DEF WHEEL Shape {
            appearance PBRAppearance {
              baseColor 0 0 0
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.02
              radius 0.05
              subdivision 24
            }
          }
        ]
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF wheelRightF HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "rightF"
        }
        PositionSensor {
          name "ps1"
        }
      ]
      endPoint Solid {
        translation -0.06 0 0.05
        rotation -9.420537450335423e-16 0 1 1.570799999999999
        children [
          USE WHEEL
        ]
        name "solid(1)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF wheelLeftR HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "leftR"
        }
      ]
      endPoint Solid {
        translation 0.06 0 -0.05
        rotation 2.355134362583857e-16 -2.355134362583857e-16 1 1.5707999999999984
        children [
          USE WHEEL
        ]
        name "solid(2)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF wheelRightR HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "rightR"
        }
      ]
      endPoint Solid {
        translation -0.06 0 -0.05
        rotation -0.17432632940257634 0.1743269790766069 0.9691338582688404 1.6021473732885239
        children [
          USE WHEEL
        ]
        name "solid(3)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    InertialUnit {
      name "iu"
    }
    Compass {
    }
    DistanceSensor {
      translation 0 -0.01 0.135
      rotation 0 0 1 -1.5707
      name "ds_fall1"
    }
    DistanceSensor {
      translation 0 0.01 0.135
      rotation 0 0 1 1.5707
      name "ds_fall2"
    }
    DistanceSensor {
      translation 0 0.01 0.135
      rotation 0 1 0 -1.5707
      name "ds_front"
      lookupTable [
        0 0 0
        0.8 2000 0
      ]
    }
    DistanceSensor {
      translation 0.05 0 0
      name "ds_left"
      lookupTable [
        0 0 0
        0.8 2000 0
      ]
    }
    DistanceSensor {
      translation -0.05 0 0
      rotation 0 1 0 3.1414
      name "ds_right"
      lookupTable [
        0 0 0
        0.8 2000 0
      ]
    }
    Emitter {
      range 1
      channel 1
    }
    Receiver {
      channel 1
    }
  ]
  name "robot_A"
  boundingObject USE BODY_GROUP
  physics Physics {
    density -1
    mass 1
  }
  controller "communication_supervisor"
  supervisor TRUE
}
Robot {
  translation -0.625635 0.06 0.483826
  rotation 0 1 0 1.5707
  children [
    DEF BODY_GROUP Group {
      children [
        DEF BODY Shape {
          appearance PBRAppearance {
            baseColor 0.917647 0.145098 0.145098
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.1 0.05 0.2
          }
        }
        Transform {
          translation 0 0 0.12
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.917647 0.145098 0.145098
                metalness 0
              }
              geometry Box {
                size 0.02 0.02 0.04
              }
            }
          ]
        }
      ]
    }
    DEF wheelLeftF HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.06 0 0.05
      }
      device [
        PositionSensor {
          name "ps"
        }
        RotationalMotor {
          name "leftF"
        }
      ]
      endPoint Solid {
        translation 0.06 0 0.05
        rotation 0 1.5700895750559044e-16 1 1.5707999999999989
        children [
          DEF WHEEL Shape {
            appearance PBRAppearance {
              baseColor 0 0 0
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.02
              radius 0.05
              subdivision 24
            }
          }
        ]
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF wheelRightF HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "rightF"
        }
        PositionSensor {
          name "ps1"
        }
      ]
      endPoint Solid {
        translation -0.06 0 0.05
        rotation -9.420537450335423e-16 0 1 1.570799999999999
        children [
          USE WHEEL
        ]
        name "solid(1)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF wheelLeftR HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "leftR"
        }
      ]
      endPoint Solid {
        translation 0.06 0 -0.05
        rotation 2.355134362583857e-16 -2.355134362583857e-16 1 1.5707999999999984
        children [
          USE WHEEL
        ]
        name "solid(2)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF wheelRightR HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "rightR"
        }
      ]
      endPoint Solid {
        translation -0.06 0 -0.05
        rotation -0.17432632940257634 0.1743269790766069 0.9691338582688404 1.6021473732885239
        children [
          USE WHEEL
        ]
        name "solid(3)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    InertialUnit {
      name "iu"
    }
    Compass {
    }
    DistanceSensor {
      translation 0 -0.01 0.135
      rotation 0 0 1 -1.5707
      name "ds_fall1"
    }
    DistanceSensor {
      translation 0 0.01 0.135
      rotation 0 0 1 1.5707
      name "ds_fall2"
    }
    DistanceSensor {
      translation 0 0.01 0.135
      rotation 0 1 0 -1.5707
      name "ds_front"
      lookupTable [
        0 0 0
        0.8 2000 0
      ]
    }
    DistanceSensor {
      translation 0.05 0 0
      name "ds_left"
      lookupTable [
        0 0 0
        0.8 2000 0
      ]
    }
    DistanceSensor {
      translation -0.05 0 0
      rotation 0 1 0 3.1414
      name "ds_right"
      lookupTable [
        0 0 0
        0.8 2000 0
      ]
    }
    Emitter {
      range 1
      channel 1
    }
    Receiver {
      channel 1
    }
  ]
  name "robot_B"
  boundingObject USE BODY_GROUP
  physics Physics {
    density -1
    mass 1
  }
  controller "communication_supervisor_B"
  supervisor TRUE
}
